### ISSUE TITLE
Wrong number type on Java/C# with inline

### ISSUE THREAD

**Issue by RblSb** [HaxeFoundation/haxe#8220]
```haxe
class Main {

	static function main():Void {
		var height = 100;
		breaker(height);
	}

	static inline function breaker(height:Float):Void {
		trace(1.0 / -height); // -0.01
		trace(1 / -height); // 0, missed "(double)" for "1"
	}

}
```
haxe git/rc1, hxjava from haxelib

**Comment by RobDangerous** [comment]
Want to point out the utter viciousness of this issue, it semi-randomly injects hard to find bugs into the code.

**Comment by nadako** [comment]
Ouch, this is bad, would be nice to fix for 4.0. Probably some missing case in some gencommon filter :)

The AST is interesting too: the `-height` unop is `Float`, but `height` itself is `Int`.
```
[Binop:Float]
	[Const:Int] 1
	/
	[Unop:Float]
		-
		Prefix
		[Local height(18440):Int:Int]

```
althought if we change it to `1 / height`, then we get this (and it also works correctly):
```
[Binop:Float]
	[Const:Int] 1
	/
	[Local height(19410):Float:Float]
```
